x-pack/plugins/apm/server/lib/services/get_service_dependencies/index.ts

```diff
@@ -0,0 +1,203 @@
+        Omit<ValuesType<typeof metrics>, 'key'>
+      >(
+        (prev, current) => {
+          return {
+            value: {
+              count: prev.value.count + current.value.count,
+              latency_sum: prev.value.latency_sum + current.value.latency_sum,
+              error_count: prev.value.error_count + current.value.error_count,
+            },
+            timeseries: joinByKey(
+              [...prev.timeseries, ...current.timeseries],
+              'x',
+              (a, b) => ({
+                x: a.x,
+                count: a.count + b.count,
+                latency_sum: a.latency_sum + b.latency_sum,
+                error_count: a.error_count + b.error_count,
+              })
+            ),
+          };
+        },
+        {
+          value: {
+            count: 0,
+            latency_sum: 0,
+            error_count: 0,
+          },
+          timeseries: [],
+        }
+      );
+
+      const deltaAsMinutes = (end - start) / 60 / 1000;
+
+      const destMetrics = {
+        latency: {
+          value:
+            mergedMetrics.value.count > 0
+              ? mergedMetrics.value.latency_sum / mergedMetrics.value.count
+              : null,
+          timeseries: mergedMetrics.timeseries.map((point) => ({
+            x: point.x,
+            y: point.count > 0 ? point.latency_sum / point.count : null,
+          })),
+        },
+        throughput: {
+          value:
+            mergedMetrics.value.count > 0
+              ? mergedMetrics.value.count / deltaAsMinutes
+              : null,
+          timeseries: mergedMetrics.timeseries.map((point) => ({
+            x: point.x,
+            y: point.count > 0 ? point.count / deltaAsMinutes : null,
+          })),
+        },
+        error_rate: {
+          value:
+            mergedMetrics.value.count > 0
+              ? (mergedMetrics.value.error_count ?? 0) /
+                mergedMetrics.value.count
+              : null,
+          timeseries: mergedMetrics.timeseries.map((point) => ({
+            x: point.x,
+            y: point.count > 0 ? (point.error_count ?? 0) / point.count : null,
+          })),
+        },
+      };
+
+      if ('service' in destination) {
+        return {
+          name: destination.service!.name,
+          serviceName: destination.service!.name,
+          environment: destination.service!.environment,
+          agentName: destination.service!.agentName,
+          ...destMetrics,
+        };
+      }
+
+      return {
+        name: destination[SPAN_DESTINATION_SERVICE_RESOURCE],
+        spanType:
+          'span.type' in destination ? destination[SPAN_TYPE] : undefined,
+        spanSubtype:
+          'span.subtype' in destination ? destination[SPAN_SUBTYPE] : undefined,
+        ...destMetrics,
+      };
+    }
+  );
+
+  const latencySums = metricsByResolvedAddress
+    .map((metrics) => metrics.latency.value)
+    .filter((n) => n !== null) as number[];
```

Comment:
nit: To avoid the explicit type you can use our existing helper that has a type guard:

```suggestion
    .filter(isValidCoordinateValue)
```

You should probably rename it though
```suggestion
    .filter(isNumber)
```

Reply:
Does that work with `filter` though?

Reply:
Yes, type guards work in filter and will return a nice clean array without nullables :)